specs/ReleaseChannelPreferenceSet.md

```diff
@@ -0,0 +1,247 @@
+RuntimeChannelPreferenceSet.md
+===
+
+# Background
+The [WebView2 Runtime](https://developer.microsoft.com/en-us/microsoft-edge/webview2/) is based on
+the evergreen Chromium platform that receives monthly major updates. You should ensure that your
+app works well with incoming browser changes by testing regularly against the WebView2 prerelease
+builds, which ship along with the [Edge preview channels](https://www.microsoft.com/en-us/edge/download/insider?form=MA13FJ):
+Canary (daily), Dev (weekly), and Beta (every 4 weeks). Prerelease testing can help you catch
+issues before they reach the stable WebView2 Runtime and affect end users.
+
+## Channel overview
+
+|Channel|Primary purpose|How often updated with new features|
+|:---:|---|:---:|
+|Stable (WebView2 Runtime)|Broad Deployment|~4 weeks|
+|Beta|Flighting with inner rings, automated testing|~4 weeks|
+|Dev|Automated testing, selfhosting to test new APIs and features|Weekly|
+|Canary|Automated testing, selfhosting to test new APIs and features|Daily|
+
+Currently you can test against prerelease channels by setting the `RuntimeChannelPreference` or
+`BrowserExecutableFolder` overrides, described in [Test upcoming APIs and features](https://learn.microsoft.com/en-us/microsoft-edge/webview2/how-to/set-preview-channel).
+These overrides are useful for testing experimental APIs but are not designed to support more
+complex automated testing and flighting scenarios.
+
+To make it easier to implement prerelease testing, the WebView2 team is adding new environment
+options for channel selection, `RuntimeChannelPreferenceSet` and `ShouldReverseChannelSearchOrder`.
+These options can be used together to specify which channels the WebView2 loader should search for
+in order of least to most stable: Canary -> Dev -> Beta -> Stable WebView2 Runtime.
+For example, you can set `ShouldReverseChannelSearchOrder` to true and specify {Beta, Stable}
+to tell the WebView2 loader to try running the app with Beta channel if it exists on the machine,
+and otherwise fall back on the stable WebView2 Runtime. With the new options you can:
+- Run automated tests against different channels on a machine with multiple channels installed
+- Target a particular channel when flighting
+- Apply different channels to different flighting rings
+- Fall back on a more stable preview channel
+
+# Examples
+## RuntimeChannelPreferenceSet
+The runtime channel preference set and channel search order are specified on the
+CoreWebView2EnvironmentOptions prior to creating the WebView2 environment.
+```c#
+CoreWebView2Environment _webViewEnvironment;
+async System.Threading.Tasks.Task CreateEnvironmentAsync() {
+    CoreWebView2RuntimeChannel channels = (CoreWebView2RuntimeChannel)
```

Comment:
Do we have real examples that we want to show in the sample code here? For example, your doc about flighting - maybe a check to see if we should be flighting or not and setting the options as appropriate?